pintc/src/expr/evaluate.rs

```diff
@@ -366,13 +391,34 @@ impl Evaluator {
                         }
                     }
 
+                    // Union variants can be cast to `int` or `real` but only if the union is an
+                    // enumeration
+                    Imm::UnionVariant {
+                        tag_num,
+                        ref ty_path,
+                        ..
+                    } => {
+                        let is_enumeration_union = Type::Union {
+                            path: ty_path.clone(),
+                            span: empty_span(),
+                        }
```

Comment:
Creating the type like this from `ty_path` is not great and I think we're doing that somewhere else already so I did the same... but that should be improved.

Reply:
Yep, I'm keen now to have the union decls referred to by `slotmap` key instead of the crappy string matching the `Custom` types (and similar) do now.